.github/workflows/rust.yml

```diff
@@ -68,6 +71,12 @@ jobs:
       - name: Check workspace without default features
         run: cargo check --no-default-features -p datafusion
 
+      - name: Check tests in each package
+        run:  |
+            for package in $(cargo metadata --no-deps --format-version 1 | jq -r '.packages[].name'); do
+              cargo check --tests --package "$package" || exit 1
+            done
```

Comment:
I am not sure if we need `--no-default-features` to check tests 🤔

Reply:
I think this command verifies that `cargo test --package "$package"` will compile successfully, but it doesn't check if `cargo test --no-default-features --package "$package"`  would run successfully

But I am not sure how common running `cargo test --no-default-features --package "$package"` is 🤔